CMakeLists.txt

```diff
@@ -0,0 +1,70 @@
+cmake_minimum_required(VERSION 3.20)
+
+project(wxcharts
+        VERSION 1.1
+        LANGUAGES CXX)
+
+################
+# config cmake #
+################
+
+set(CMAKE_CXX_STANDARD 17 CACHE STRING "C++ Standard")
+set(CMAKE_CXX_STANDARD_REQUIRED ON CACHE BOOL "")
+set(CMAKE_CXX_EXTENSIONS OFF CACHE BOOL "")
+set(CMAKE_DEBUG_POSTFIX "-d" CACHE STRING "Postfix for debug output")
+set(wxWidgets_ROOT_DIR "" CACHE PATH "")
+set(wxWidgets_wxrc_EXECUTABLE "" CACHE FILEPATH "")
+
+
+option(BUILD_SHARED_LIBS "Build shared libraries" ON)
+
+#############
+#dependency #
+#############
+
+include(GNUInstallDirs)
+
+find_package(wxWidgets REQUIRED COMPONENTS core base)
+
+include(${wxWidgets_USE_FILE})
+
+##########
+# target #
+##########
+
+file(GLOB SOURCE_FILES src/*.cpp)
+
+add_library(${PROJECT_NAME} ${SOURCE_FILES})
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
+
+# export macro
+include(GenerateExportHeader)
+generate_export_header(${PROJECT_NAME})
+
+target_include_directories(${PROJECT_NAME}
+                           PUBLIC # use PUBLIC instead of PRIVATE because tests use it
```

Comment:
What does this do?

Reply:
It is used to generate macro `WXCHARTS_EXPORT ` in a portable way, DLL on Windows needs it to export classes and functions, example as below

```
class WXCHARTS_EXPORT wxAreaChartDataset
```